### PR TITLE
INFR-1737 no-release stages do not produce releases / are invisible from dashbo…

### PR DESCRIPTION
**Pull request by grosser**
…ards

atm build stages show in the dashboards, which they should not, since they are not code releases.
This introduces a `release` flag for the deploy, so we know it is a proper code deploy and not some random other tasks like building assets

... in the next PR: kill macros and make them stages that do not generate releases

https://zendesk.atlassian.net/browse/INFR-1737

@zendesk/runway 

screen shot 2016-03-17 at 2 28 12 pm

### Risks
 - Low: deploys failing because db was not migrated
 - Low: users being confused because build stages are missing from the dashboards

### CODE REVIEW

**Comment by zendesknotifications** [comment]
&nbsp;<b>Code Climate</b> has <a href="https://codeclimate.com/repos/53340bef6956800b9000675c/pull/763">analyzed this pull request</a>.

**Comment by grosser** [comment]
@sandlerr can you review this one too ?

File: app/models/stage.rb [hunk 1]
```diff
@@ -98,7 +98,7 @@ def confirm_before_deploying?
   end
 
   def create_deploy(user, attributes = {})
-    deploys.create(attributes) do |deploy|
+    deploys.create(attributes.merge(release: !no_code_deployed)) do |deploy|
```
Comment (sandlerr) [inline app/models/stage.rb]: or even `attributes.merge(release: !attributes.delete(:no_code_deployed))`

**Comment by sandlerr** [comment]
I'm not sure if these really are the same thing - this is wrapped in `BuddyCheck.enabled?` but now it provides a function that could be useful for orgs that don't use the buddy check. Do you think it could also have some use outside of production?

File: app/models/stage.rb (hunk 1 above)
Comment (grosser) [inline app/models/stage.rb]: Why would we need to pass in the `:no_code_deployed` ... a stage is always the same command, so it's always either a deploy or not a deploy

**Comment by grosser** [comment]
unwrapped from BuddyCheck ... since it's useful with and without.

**Comment by sandlerr** [comment]
:+1: on green

**Pull request merged**